pandas/core/tools/numeric.py

```diff
@@ -275,7 +275,8 @@ def to_numeric(
     # GH33013: for IntegerArray, BooleanArray & FloatingArray need to reconstruct
     # masked array
     if (mask is not None or new_mask is not None) and not is_string_dtype(values.dtype):
-        if mask is None:
+        if mask is None or (new_mask is not None and errors == "coerce"):
```

Comment:
Can't really think about a case right now, but I think ``new_mask.shape == mask.shape`` instead of ``errors="coerce"`` is safer

Reply:
Sure I'll change it to that